### PR TITLE
Fixes #1371 and #1578

### LINKED ISSUE #1371: satisfaction survey missing
GLPI 9.2

Hello

some users are not able to fill the satisfaction suvey, it is strange because they receive the mail but they don't have the approval for satisfaction survey.

Do you have an idea to fix this issue.

just in self service profile it does not work anymore

I have disabled all plugins but same behavior

### LINKED ISSUE #1578: Review Ticket::SURVEY right
Implementation seems broken

According to original ticket #340 this new right should only apply to ticket author (right still always granted to requester).

So in central.class.php this part of code have no sense: 

```
         if ($showsurvey) {
            Ticket::showCentralList(0, "survey", false);
         }
```

in showCentralList, "survey" list must be fixed (request criteria)
Ticket::getTabNameForItem
TicketSatisfaction::canUpdateItem
etc...

### PR DESCRIPTION
**Pull request by tomolimo**
Changed way of testing rights for survey when logged in user is also requester

Fixes #1371
Fixes #1578

### CODE REVIEW

File: inc/ticket.class.php [hunk 1]
```diff
@@ -610,12 +610,11 @@ function getTabNameForItem(CommonGLPI $item, $withtemplate=0) {
...
-            if (($item->fields['status'] == self::CLOSED)
-                && Session::haveRight('ticket', Ticket::SURVEY)){
-               $satisfaction = new TicketSatisfaction();
-               if ($satisfaction->getFromDB($item->getID())) {
-                  $ong[3] = __('Satisfaction');
-               }
+            $satisfaction = new TicketSatisfaction();
+            if ($satisfaction->getFromDB($item->getID())
+                     && $item->fields['status'] == self::CLOSED
+                     && $satisfaction->canUpdateItem()){
```
Comment (remicollet) [inline inc/ticket.class.php]: In fact, not sure about this one.
I don't think you need the right to "update" (answer) the survey to see it.

Only removing the Session::haveRight('ticket', Ticket::SURVEY) should be enough

Comment (tomolimo) [inline inc/ticket.class.php]: I made a test without the canUpdateItem() condition, and it seems ok

Comment (tomolimo) [inline inc/ticket.class.php]: Yes, it's enough, because of the showFormButtons() which will test if can UPDATE, and it will prevent to save the form.

Comment (tomolimo) [inline inc/ticket.class.php]: And it's even better, as like this the form can be viewed event if the right to update is not given

**Comment by orthagh** [comment]
Also ok

**Pull request merged**